subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -695,7 +697,7 @@ int nrf_cloud_state_encode(uint32_t reported_state, const bool update_desired_to
 		ret += json_add_null_cs(pairing_obj, NRF_CLOUD_JSON_KEY_CFG);
 		ret += json_add_null_cs(reported_obj, NRF_CLOUD_JSON_KEY_STAGE);
 		ret += json_add_null_cs(reported_obj, NRF_CLOUD_JSON_KEY_TOPIC_PRFX);
-		ret += json_add_null_cs(connection_obj, NRF_CLOUD_JSON_KEY_KEEPALIVE);
+		ret += shadow_connection_reset(connection_obj);
```

Comment:
What would happen if we called `shadow_connection_update` here instead of `shadow_connection_reset`?

Reply:
Why?  The original code here set all fields to null.  I do not see a reason to break the pattern.

Reply:
My thinking was it would let you get rid of the `shadow_connection_reset` function altogether

And thinking in terms of how BE processes this stuff, writing blank here before later writing the actual data doesn't seem like it serves much purpose, but possibly I am mistaken

Reply:
Sure, in that case you can still consider keeping `ret += json_add_null_cs(connection_obj, NRF_CLOUD_JSON_KEY_KEEPALIVE);` if you want to avoid `shadow_connection_reset`, but if we're planning to rewrite all this it isn't a huge deal to me either way

Reply:
But, the point of setting the JSON keys to a null value is, if they do not get set to a non-null value, it removes the key and value from the shadow.  It's basically a way to delete a field in the shadow.

If the UA_PIN_WAIT state is reached, the fields are nulled.  If it never advances to the UA_PIN_COMPLETE state, the fields will be deleted in the shadow.  Otherwise they will be set to the proper values in the shadow.

Reply:
i believe the values are null'd just to sanitize the shadow so it is in a known state before continuing.

but i think you can just set the entire `"connection"` object to NULL, instead of the stuff inside it.

Reply:
The other fields set to null in nrf_cloud_state_encode() for STATE_UA_PIN_WAIT set every subfield to null.  I am only following the existing pattern here.

I separated out the shadow_connection_reset() stuff because I thought I might need it for REST and CoAP, but I do not.  So I can just remove this function and put the json_add_null_cs() calls in nrf_cloud_state_encode().

Reply:
> But, the point of setting the JSON keys to a null value is, if they do not get set to a non-null value, it removes the key and value from the shadow

I guessed that might be the intent, but I will point out that we wind up writing several of them blank in the `STATE_UA_PIN_COMPLETE` case if they aren't set, so writing them blank in `STATE_UA_PIN_WAIT` still comes off as pointless to me.

A couple of them I'm not super sure on the behavior of, but I'd put forward that in a hypothetical cleanup of this area, maybe instead of blanking first, and then writing values later, let's consider writing a function that just writes all device info and blanks anything not in use by the device in one fell swoop.

Reply:
https://nordicsemi.atlassian.net/browse/IRIS-6835

Reply:
I assume the author did this in STATE_UA_PIN_WAIT in case STATE_UA_PIN_COMPLETE was never reached.  But someone can figure that out when they do the above ticket.